core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -1052,11 +1052,28 @@ object IO {
    * Evaluate the elements of an `Iterable[A]` in parallel
    * and collect the results. This is the parallel version of `traverse`.
    */
-  def parTraverse[E, A, B](as: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+  final def parTraverse[E, A, B](as: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
     as.foldRight[IO[E, List[B]]](IO.sync(Nil)) { (a, io) =>
-      fn(a).par(io).map { case (b, bs) => b :: bs }
+      fn(a).parWith(io)((b, bs) => b :: bs)
     }
 
+  /**
+   * Evaluate the elements of a traversable data structure in parallel
+   * and collect the results. Only up to `n` tasks run in parallel.
+   * This is a version of `parTraverse`, with a throttle.
+   */
+  final def parTraverseThrottled[E, A, B](n: Long)(as: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+    for {
+      semaphore <- Semaphore(n)
+      bs <- parTraverse(as) { a =>
+             for {
+               _ <- semaphore.acquire
```

Comment:
This is not currently safe and will have to be wrapped using `withPermit`. Otherwise, semaphores will leak in the presence of failure (errors and interruption).

Reply:
done